homeassistant/components/shelly/config_flow.py

```diff
@@ -460,6 +477,8 @@ def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         return (
             get_device_entry_gen(config_entry) in RPC_GENERATIONS
             and not config_entry.data.get(CONF_SLEEP_PERIOD)
+            and cast(bool, config_entry.data.get(CONF_SCRIPT))
+            # Needed for devices configured before introducing CONF_SCRIPT
```

Comment:
```suggestion
```
As far as I understand this comment is misleading as wall display will need it regardless if it was configured before this PR or after

Reply:
Fixed in a future proof way